python/meep.i

```diff
@@ -1405,6 +1405,7 @@ void _get_gradient(PyObject *grad, PyObject *fields_a, PyObject *fields_f, PyObj
     if ($1.num_items != 0) {
         for (int i = 0; i < $1.num_items; i++) {
             material_free($1.items[i]);
+            delete $1.items[i];
```

Comment:
Maybe it would be better to put this in `material_free`?

Reply:
We could - but then it might break the code of anybody using the C++ api directly.
I dont know how much we worry about that.

Reply:
I'm not so worried about that.

Reply:
I pushed a patch to move `delete` into `material_free`.   It looks like there was another memory leak in `set_default_material` that this fixes.

Reply:
Thanks for making the change. I tested this internally and everything looks good.
Please go ahead and merge this change.